openhantek/src/hantekdso/hantekdsocontrol.cpp

```diff
@@ -65,6 +65,9 @@ HantekDsoControl::HantekDsoControl(USBDevice *device)
 
     // Apply special requirements by the devices model
     device->getModel()->applyRequirements(this);
+    // Initialize the gain vector with standard and safe values
+    controlsettings.probeGain.resize( specification.channels, 1);
```

Comment:
Please do this in the controlsettings constructor. But read the next comment first. It might make sense to not apply the probe gain here but in a post processing step.